spectator/http_client.cc

```diff
@@ -173,7 +177,9 @@ HttpResponse HttpClient::do_post(const std::string& url,
   auto logger = registry_->GetLogger();
   curl.set_url(url);
   curl.set_headers(headers);
-  curl.post_payload(payload, size);
+  if (strcmp("POST", method) == 0) {
```

Comment:
Could this just check if the payload size is greater than 0? That would allow it to work with `PUT` or odd things like ElasticSearch that has APIs that expect a request body with a `GET` request.

Reply:
I thought about it but it wouldn't work as is now. We'd have to refactor how we handle the request body (change from `CURLOPT_POSTFIELDS`/`CURLOPT_POSTFIELDSIZE`) to something more generic and this wasn't intended to be a generic http-client, although it's heading that way.